compiler/modules/types/enumerable.bal

```diff
@@ -165,6 +165,10 @@ function compareEnumerable(EnumerableType v1, EnumerableType v2) returns Order {
         string s2 = <string>v2;
         return v1 == s2 ? EQ : (v1 < s2 ? LT : GT);
     }
+    else if v1 is decimal {
+        decimal d2 = <decimal>v2;
+        return v1 == d2 ? EQ : (v1 < d2 ? LT : GT);
```

Comment:
Please read the spec and check if this expression works as expected in jBallerina beta2, specially for special values such as -0/NaN/Inf, if it doesn't you'll have to create to a polyfill slimier to `floatEq` below.

Reply:
Decimal doesn't have -0, infinity or nan.

Reply:
```ballerina
decimal d = 0d/0d;
io:println(d);
```
This prints `NaN` on jBallerina beta2.
But as you say, since they shouldn't exist in the first place, we shouldn't check them here then.
We should reject them at parser and (for this PR) json.parse.

Reply:
Ballerina semantics are as described here: http://speleotrove.com/decimal/dax3274.html

I don't think we invest a lot of effort on polyfills. The ultimate goal is that nBallerina compiled with itself is correct.  If there are obscure cases that fail because of jBallerina bugs, we should record those as an f case, and wait for the jBallerina team to fix them.